src/System.Linq.Expressions/src/System/Linq/Expressions/IArgumentProvider.cs

```diff
@@ -5,10 +5,10 @@
 namespace System.Linq.Expressions
 {
     /// <summary>
-    /// Provides an internal interface for accessing the arguments that multiple tree
-    /// nodes (DynamicExpression, ElementInit, MethodCallExpression, InvocationExpression, NewExpression,
-    /// and InexExpression).
-    /// 
+    /// Interface for accessing the arguments of multiple tree nodes (DynamicExpression, ElementInit,
+    /// MethodCallExpression, InvocationExpression, NewExpression, and IndexExpression).
```

Comment:
Yes, with the exception of `DynamicExpression` which lives in another assembly, but I could use the interface type there instead.